interpreter/valid/valid.ml

```diff
@@ -142,8 +142,8 @@ let check_unop unop at =
   | Values.I32 I32Op.Extend32S -> error at "invalid unary operator"
   | _ -> ()
 
-let check_memop (c : context) (memop : 'a memop) get_sz at =
-  ignore (memory c (0l @@ at));
+let check_memop (c : context) (memop : 'a memop) get_sz sh at =
```

Comment:
Nit: can you put a type annotation on the `sh` parameter, and place it before `get_sz`? It is part of the judgement that this function implements, while the latter is just an auxiliary thing like `at`.

Reply:
Done.